core/src/robot/AbstractTransmission.cc

```diff
@@ -0,0 +1,374 @@
+        if (isAttached_)
+        {
+            PRINT_ERROR("Transmission already attached to a robot. Please 'detach' method before attaching it.");
+            return hresult_t::ERROR_GENERIC;
+        }
+
+        // Make sure the robot still exists
+        if (robot.expired())
+        {
+            PRINT_ERROR("Robot pointer expired or unset.");
+            return hresult_t::ERROR_GENERIC;
+        }
+
+
+        // Copy references to the robot and shared data
+        robot_ = robot;
+
+        // Update the flag
+        isAttached_ = true;
+
+        return hresult_t::SUCCESS;
+    }
+
+    hresult_t AbstractTransmissionBase::detach(void)
+    {
+        if (!isAttached_)
+        {
+            PRINT_ERROR("Transmission not attached to any robot.");
+            return hresult_t::ERROR_GENERIC;
+        }
+
+        // Clear the references to the robot
+        robot_.reset();
+
+        // Unset the Id
+        transmissionIdx_ = -1;
+
+        // Reset constant members
+        jointNames_.clear();
+        jointModelIndices_.clear();
+        jointTypes_.clear();
+        jointPositionIndices_.clear();
+        jointVelocityIndices_.clear();
+        motorNames_.clear();
+        motorIndices_.clear();
+
+        // Update the flags
+        isAttached_ = false;
+        isInitialized_ = false;
+
+        return hresult_t::SUCCESS;
+    }
+
+    hresult_t AbstractTransmissionBase::setOptions(configHolder_t const & /*transmissionOptions*/)
+    {
+        // TODO SetOptions
+        return hresult_t::SUCCESS;
+    }
+
+    configHolder_t AbstractTransmissionBase::getOptions(void) const
+    {
+        return transmissionOptionsHolder_;
+    }
+
+    hresult_t AbstractTransmissionBase::refreshProxies(void)
+    {
+        hresult_t returnCode = hresult_t::SUCCESS;
+
+        if (!isAttached_)
+        {
+            PRINT_ERROR("Transmission not attached to any robot. Impossible to refresh proxies.");
+            returnCode = hresult_t::ERROR_INIT_FAILED;
+        }
+
+        auto robot = robot_.lock();
+        if (returnCode == hresult_t::SUCCESS)
+        {
+            if (!robot)
+            {
+                PRINT_ERROR("Robot has been deleted. Impossible to refresh proxies.");
+                returnCode = hresult_t::ERROR_GENERIC;
+            }
+        }
+
+        if (returnCode == hresult_t::SUCCESS)
+        {
+            if (!isInitialized_)
+            {
+                PRINT_ERROR("Transmission not initialized. Impossible to refresh proxies.");
+                returnCode = hresult_t::ERROR_INIT_FAILED;
+            }
+        }
+
+        if (returnCode == hresult_t::SUCCESS)
+        {
+            if (!robot->getIsInitialized())
+            {
+                PRINT_ERROR("Robot not initialized. Impossible to refresh proxies.");
+                returnCode = hresult_t::ERROR_INIT_FAILED;
+            }
+        }
+
+        jointIndex_t jointModelIdxTmp;
```

Comment:
it is here bc of this error otherwise:
```
/home/fabian.schramm/wdc_workspace/src/jiminy/core/src/robot/AbstractTransmission.cc: In member function ‘virtual jiminy::hresult_t jiminy::AbstractTransmissionBase::refreshProxies()’:
/home/fabian.schramm/wdc_workspace/src/jiminy/core/src/robot/AbstractTransmission.cc:256:34: error: declaration of ‘jointModelIdx’ shadows a previous local [-Werror=shadow]
  256 |             jointIndex_t const & jointModelIdx = robot->pncModel_.getJointId(jointName);
```

Reply:
I'm using `jointModelIdx` also somewhere else in this class, and this seems to be a problem

Reply:
Then just do `jointModelIdx = robot->pncModel_.getJointId(jointName);` at line 256 and you are good :) Recycling is good for earth :)